test/runtests.jl

```diff
@@ -226,6 +226,50 @@ else
 end # @static if VERSION < v"1.8"
     end
 
+    @testset "FIFOLock" begin
+@static if VERSION < v"1.10"
+        @warn "skipping FIFOLock tests since VERSION ($VERSION) < v\"1.10\""
+else
+        fl = FIFOLock()
+        lock(fl)
+        tot = 0
+        ttsks = Task[]
+        tordr = Int[]
+        try
+            # we assume here that the tasks spawned below run in the order
+            # they were spawned
+            for i in 1:16
+                t = Threads.@spawn begin
+                    lock(fl)
+                    try
+                        tot = tot + 1
+                        if tot != i
+                            error("non-atomic access in lock")
+                        end
+                        push!(tordr, i)
+                    finally
+                        unlock(fl)
+                    end
+                end
+                push!(ttsks, t)
+                sleep(0.1)
```

Comment:
😁 i'm going to complain about this being a flaky test again. :)

Can you use concurrency primitives to ensure these line up in the lock sequentially? It seems like it should be doable with a bit of thought. Not sure yet tho

Reply:
If you don't care about having all 16 in the queue before they proceed, it's might be okay? You can have each task spawn the next one, right before grabbing the lock. But i'm not sure if you could guarantee getting all 16 in the queue.

Reply:
Fair point. I think I've worked out a better way.